docs/navigator_api/js/streaming_navigator.js

```diff
@@ -1,85 +1,211 @@
-      } else {
-        console.log(data.data);
+    while (true) {
+      let iterateResult;
+      try {
+        iterateResult = await readFromStream({ stream_id, num_rows, iterator });
+      } catch (err) {
+        console.log(`Error iterating on stream: ${JSON.stringify(err)}`);
+        break;
       }
+
+      iterator = iterateResult.next_iterator;
+
+      for (const data of iterateResult.data) {
+        if (data.data_type === "TabularResponse") {
+          rowHandler(data.data);
+          resultCount++;
+        } else {
+          console.log(data.data);
+        }
+      }
+      if (
+        iterateResult.stream_state.status === "closed" ||
+        resultCount >= num_rows
+      ) {
+        break;
+      }
+      await new Promise((r) => setTimeout(r, POLLING_DELAY));
     }
-    if (
-      iterateResult.stream_state.status === "closed" ||
-      resultCount >= num_rows
-    ) {
-      break;
-    }
-    await new Promise((r) => setTimeout(r, 500));
-  }
+  };
+
+  // Combines functions above to creates a single stream and then fetch data from that stream.
+  // Currently, each stream can return a maximum of 50 rows.
+  const createStreamAndMakeData = async ({
+    model_id,
+    num_rows,
+    prompt,
+    table_headers,
+    table_data,
+    params,
+    rowHandler,
+  }) => {
+    const { stream_id } = await createStream({
+      model_id,
+      num_rows,
+      prompt,
+      table_headers,
+      table_data,
+      params,
+    });
+
+    return await getDataFromStream({ stream_id, num_rows, rowHandler });
+  };
+
+  /**
+   * Since each individual stream allows a MAX of 50 rows, create multiple streams as needed,
+   * passing results of the previous stream into the next stream.
+   */
+  const recursiveStreaming = async ({
+    numRows,
+    prompt,
+    table_headers,
+    table_data,
+  }) => {
+    let numRowsLeft = numRows;
+    /**
+     * gather generated data from current streaming to pass to
+     * the next stream.
+     */
+    let results = [];
+
+    const newRowHandler = (row) => {
+      numRowsLeft--;
+      // add to internal tracker
+      results.push(row);
+      // call external callback with current data
+      rowCallback(row);
+    };
+
+    const numRowsForStream =
+      numRows > MAX_ROWS_PER_STREAM ? MAX_ROWS_PER_STREAM : numRows;
+
+    return createStreamAndMakeData({
+      model_id,
+      num_rows: numRowsForStream,
+      prompt: prompt,
+      params,
+      table_data,
+      table_headers,
+      rowHandler: newRowHandler,
+    }).then(() => {
+      if (numRowsLeft > 0 && results.length) {
+        const table_headers = results[0].table_headers;
+        const tableHeadersString = table_headers.join(", ");
+        const table_data = results.map((row) => row.table_data).flat();
+
+        try {
+          return recursiveStreaming({
+            numRows: numRowsLeft,
+            prompt: `Generate more data like the following table with the columns: ${tableHeadersString}`,
```

Comment:
Where is this prompt from?

Reply:
So I believe if we include the previous rows in the right API parameter (ref_data.sample_data) then the backend handles this for us and we can just pass the original prompt here. [Here](https://github.com/gretelai/gretel-python-client/blob/main/src/gretel_client/inference_api/tabular.py#L186) is the matching python logic. We'll need to add these new params to createStream above as well.

Reply:
DM'd you to understand what the correct api is, since the other usage examples I have conflict.